app/src/main/java/android/example/wordlehelper/StatsManager.kt

```diff
@@ -20,10 +28,60 @@ class StatsManager(context: Context) {
         return Gson().fromJson(userStatsString, UserStats::class.java)
     }
 
-    fun storeStats(data: UserStats) {
+    private fun storeStats(data: UserStats) {
         val userStatsJson: String = Gson().toJson(data)
         file.writeText(userStatsJson)
     }
+
+     fun onGiveUpUpdateStats() {
+        val userStats = getStats() ?: UserStats()
+        userStats.timesPlayed++
+        userStats.timesGivenUp++
+        storeStats(userStats)
+    }
+
+    fun defeatStatsUpdate(context: Context, attempt: LinearLayout, guessWordsLayout: LinearLayout){
+        val statsFile = File(context.filesDir, "statsFile.json")
+        if (statsFile.exists()) {
+            val stats = readStatsFromFile(statsFile)
+            stats.averageTries =
+                (((stats.timesPlayed * stats.averageTries!!) + (guessWordsLayout.indexOfChild(
+                    attempt
+                ) + 1)) / (stats.timesPlayed + 1))
+            stats.timesPlayed++
+            writeStatsFile(statsFile, stats)
+        } else {
+            val data = UserStats()
+            data.timesPlayed = 1
+            data.averageTries =
+                (guessWordsLayout.indexOfChild(attempt) + 1).toFloat()
+            data.timesGivenUp = 0
+            data.timesWon = 0
+            writeStatsFile(statsFile, data)
+        }
+    }
+
+    fun victoryStatsUpdate(context: Context, attempt: LinearLayout, guessWordsLayout: LinearLayout){
```

Comment:
This function is almost the same as the previous one. I would refactor them to avoid duplication.